.gdbinit

```diff
@@ -1139,6 +1139,17 @@ location, if available. Optionally list the frame arguments and locals too."""
             name = elem.sym
             equal = ansi('=', R.style_low)
             value = to_string(elem.sym.value(frame))
+            if (elem.sym.type.code in (gdb.TYPE_CODE_REF,
+                                       gdb.TYPE_CODE_RVALUE_REF)):
+                try:
+                    ref_value = to_string(
+                        elem.sym.value(frame).referenced_value())
+                except gdb.MemoryError:
+                    ref_value = None
+
+                if ref_value and not ref_value == value:
```

Comment:
I noticed that for some types `value = to_string(elem.sym.value(frame))` already evaluates to the referenced value (e.g. for `string&`), so this is just to avoid unnecessary redundancy.

Reply:
I cannot reproduce this, without this patch `rx` is displayed as `loc rx = @0x7fffffffe7f0` in the following program even though it is a `std::string &`:

```c++
#include <iostream>
#include <string>

int main() {
    std::string x = "foo";
    std::string &rx = x;
    std::cout << rx << std::endl;
}
```

What you say happens if you print the value (`p rx`) but not via the Python API.

Reply:
8.0 here, also the string itself is displayed without pretty-printers:

```
loc x = {
  static npos = 18446744073709551615,
  _M_dataplus = {
    <std::allocator<char>> = {
      <__gnu_cxx::new_allocator<char>> = {<No data fields>}, <No data fields>},
    members of std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >::_Alloc_hider:
    _M_p = 0x7fffffffe830 "foo"
  },
  _M_string_length = 3,
  {
    _M_local_buf = "foo", '\000' <repeats 12 times>,
    _M_allocated_capacity = 7303014
  }
}
loc rx = @0x7fffffffe820
```

I think this is a somewhat broken installation on my side. The system version of GDB (Debian 7.12-6) works as expected but does not support `gdb.TYPE_CODE_RVALUE_REF`.